### ISSUE TITLE
Papers, articles, videos

### ISSUE THREAD

**Issue by zakajd** [zakajd/voice-transfer#1]
Here is a long list of all relevant and not so relevant materials, which can be found in the internet. 
I will try to add a short description or a short summary for all of them, but everyone else is welcome to help me with that.

**Comment by zakajd** [comment]
# Papers
- https://arxiv.org/pdf/1802.06006.pdf
- https://arxiv.org/abs/1905.05879 
- SampleRNN: An Unconditional End-to-End Neural Audio Generation Model.
- Char2Wav: End-to-end Speech Synthesis.
- ObamaNet: Photo-realistic lip-sync from text.
- MelGAN: Generative Adversarial Networks for Conditional Waveform Synthesis
- arxiv.org/pdf/1802.06006.pdf
- https://arxiv.org/abs/1802.06840 paper by Yang Gao introduces VoiceGANs, which are capable of converting the gender of a speaker's voice.
- https://arxiv.org/abs/1806.04558
- https://arxiv.org/pdf/1711.00937.pdf vqvae
- https://arxiv.org/pdf/1906.00446.pdf vqvae  paper update, mostly focused on images
- https://arxiv.org/abs/1905.05879
- https://arxiv.org/abs/1910.03713
- https://arxiv.org/abs/1710.11385
# Articles 
- www.descript.com/post/how-imputations-work-the-research-behind-overdub
- www.kecl.ntt.co.jp/people/kaneko.takuhiro/projects/cyclegan-vc/
- https://deepmind.com/blog/article/wavenet-generative-model-raw-audio
- https://towardsdatascience.com/voice-translation-and-audio-style-transfer-with-gans-b63d58f61854
- https://avdnoord.github.io/homepage/vqvae/ TOP
- https://papers.nips.cc/paper/7700-transfer-learning-from-speaker-verification-to-multispeaker-text-to-speech-synthesis

# Videos
https://www.youtube.com/watch?v=0sR1rU3gLzQ
https://www.youtube.com/watch?v=nNCzWaFKmDk
https://youtu.be/WyU-gJ2NTGc
https://youtu.be/6KHSPiYlZ-U

# Repositories
- https://github.com/andabi/deep-voice-conversion
- https://github.com/mkotha/WaveRNN
- https://github.com/CorentinJ/Real-Time-Voice-Cloning
- https://github.com/Rayhane-mamah/Tacotron-2
- https://github.com/CorentinJ/Real-Time-Voice-Cloning/issues/30
- https://github.com/mazzzystar/randomCNN-voice-transfer
- https://github.com/andabi/deep-voice-conversion/issues
- https://github.com/liusongxiang/StarGAN-Voice-Conversion ICLR 2020
- https://github.com/unilight/cdvae-vc
- https://github.com/zalandoresearch/pytorch-vq-vae
- https://github.com/ritheshkumar95/pytorch-vqvae
- https://github.com/r9y9/deepvoice3_pytorch

# Other
www.respeecher.com
www.descript.com/lyrebird-ai
https://audiostyletransfer.wordpress.com

A new large-scale corpus of English speech derived for TTS;
LibriTTS: A Corpus Derived from LibriSpeech for Text-to-Speech
Dataset: http://openslr.org/60/
Paper: http://arxiv.org/abs/1904.02882

Most of the progress I've seen has been on speech-to-speech rather than singing-to-singing, and the most successful approaches have been (effectively) a full encoder/decoder stack. The Kate Winslet project is a good example of this (note that they train a single output speaker). Projects like Lyrebird which allow arbitrary speakers are (afaik) training a single generator that can be conditioned on a speaker vector, then learning the speaker vector. In principle, I don't see a reason why these generators can't also be conditioned on pitch information, so I would guess that full encoding / resynth is the endgame for singing style transfer. There are other possible approaches (currently messing around with some!) but I'm not optimistic. Every image-to-image (spectrogram-to-spectrogram) method I've seen has not worked very well, or only worked for very limited use-cases. The features that you need to match on are nonlocal in spectrogram space and global coherence of the output really matters. So I don't think any direct application of a popular image-to-image style transfer method will succeed.

**Comment by zakajd** [comment]
www.descript.com/post/how-imputations-work-the-research-behind-overdub

Article about text-conditioned audio-inpainting. This is not exactly related to our task, because a) they generate only small portion of audio file (the exact word/words), leaving all other parts mostly unchanged b) there is additional text conditioning, that rules the process.

There a bunch of interesting idea, but they are discussed in a very broad terms. 
1. Use bidirectional approach to give information not only about prefix audio, but also about suffix
2. Style net additionaly conditions generator with 2 vectors of fixed length, which describe "style" (intonation, speed, accent etc.) of prefix and suffix, which allows generator to match this parameters. 

3. At the very end there is a description of approach without attention:

>   First, a duration predictor network is used to predict the duration of each phoneme in the utterance to be generated. This is accomplished by a feedforward network, which attends to various conditioning information (prefix text, suffix audio, suffix text, and prefix audio) as available. A softmax layer is applied at the very end to predict each phoneme’s duration. 

> Then, another network, composed of a feedforward feature extractor and a final autoregressive layer is used to predict the audio sequence.

> Between the networks, we introduced an upsampling layer, which upsamples the text sequence to be imputed to match the audio sequence. During training, the ground truth alignment between text and audio is used to carry out the upsampling. During generation, the predicted alignment (obtained from the prediction of each phoneme’s duration) is used to carry out the upsampling.

**Comment by ilya16** [comment]
An extensive overview of voice conversion task, including terminology, classification of systems, approaches, example applications and challenges:
[https://shamidreza.github.io/papers/mohammadi_2017_overview.pdf](https://shamidreza.github.io/papers/mohammadi_2017_overview.pdf)

Good source of information for Introduction and other parts of the Report.
